src/H5Ofill.c

```diff
@@ -713,32 +707,22 @@ H5O__fill_old_size(const H5F_t H5_ATTR_UNUSED *f, const void *_fill)
 herr_t
 H5O_fill_reset_dyn(H5O_fill_t *fill)
 {
-    hid_t  fill_type_id = -1;      /* Datatype ID for fill value datatype when reclaiming VL fill values */
-    herr_t ret_value    = SUCCEED; /* Return value */
+    herr_t ret_value = SUCCEED; /* Return value */
 
     FUNC_ENTER_NOAPI(FAIL)
 
     assert(fill);
 
     if (fill->buf) {
         if (fill->type && H5T_detect_class(fill->type, H5T_VLEN, false) > 0) {
-            H5T_t *fill_type;  /* Copy of fill value datatype */
             H5S_t *fill_space; /* Scalar dataspace for fill value element */
 
-            /* Copy the fill value datatype and get an ID for it */
-            if (NULL == (fill_type = H5T_copy(fill->type, H5T_COPY_TRANSIENT)))
```

Comment:
No longer need to copy this datatype since we're not creating an ID and `H5Treclaim` won't modify it

Reply:
I think there are also several more opportunities for removing type copying calls since a lot of them only appear to have been done so that when the ID they were wrapped in is closed, it doesn't free the original datatype. But I didn't want to go down that rabbit hole yet since it could get complicated, especially in the case where a copy of a vlen or reference type is made and the "location" of the copied type is then changed.

Reply:
Agree, I mentioned this in another comment, and I think there's good opportunity for more performance improvements from removing the unnecessary copy & free calls.